exporter/datadogexporter/internal/hostmetadata/host.go

```diff
@@ -27,7 +29,7 @@ var _ = featuregate.GlobalRegistry().MustRegister(
 	featuregate.WithRegisterToVersion("0.75.0"),
 )
 
-func GetSourceProvider(set component.TelemetrySettings, configHostname string) (source.Provider, error) {
+func GetSourceProvider(set component.TelemetrySettings, configHostname string, timeout time.Duration) (source.Provider, error) {
```

Comment:
Since we are changing the `GetSourceProvider` function signature anyway, we may want to rethink it to avoid breaking every caller each time we need to add more functionality. For example, we could pass a `Config` struct with the config hostname and timeout, so we can add more optional arguments easily. Do you think that would be a good idea?

Reply:
We could but honestly my instinct is to leave it.  

The changes are pretty localized there aren't many callsites and none are outside of the datadog exporter (im including `factory.SourceProvider` in this calculus).  I think if we decide we want to customize the providers more for whatever reason (entirely possible given how many different external services this tries to hit) we should probably convert to a struct with more settings.  

Not a big deal at all - happy to change my mind - but yeah those were my thoughts..

Reply:
> none are outside of the datadog exporter

Yeah well, that's a given since this is an internal package :smile: Okay, we can leave it for later then :)